lib/src/ovis_log/ovis_log.c

```diff
@@ -469,15 +469,16 @@ static int __get_log_info(ovis_log_t l, struct ovis_log_buf *buf)
 	char *level_s;
 
 	if (l->level == OVIS_LDEFAULT)
-		level_s = "default";
+		level_s = strdup("default");
 	else
 		level_s = ovis_log_level_to_str(l->level);
 
 	rc = __buf_append(buf, "{\"name\":\"%s\","
 			        "\"desc\":\"%s\","
 			        "\"level\":\"%s\"}",
 				l->name, l->desc,
-				level_s);
+				level_s ? level_s : "out_of_memory");
```

Comment:
So your expectation is that this buffer append will succeed after a previous memory allocation has just failed?

Reply:
The buffer that the is being appended here is largish to start with, and unlikely to require reallocation while being appended with the small format string in use here.